### PR TITLE
adding Dockerise and terraform ecs service

### PR DESCRIPTION
**Pull request by myadavch**

### CODE REVIEW

File: terraform/groups/ecs-service/locals.tf [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Define all hardcoded local variable and local variables looked up from data resources
+locals {
+  stack_name                = "company-requests" # this must match the stack name the service deploys into
```
Comment (amartin7663) [inline terraform/groups/ecs-service/locals.tf:3]: Is this the correct cluster? Looking at the page it isn't actually in one though accounts filing api is in filing maintain: https://companieshouse.atlassian.net/wiki/spaces/Arch/pages/4217208873/ECS+Cluster+Architecture

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+allowed_company_prefixes = "SC,NI,OC,SO,R,AP"
+
+cache_server = "redis"
+cdn_host = "d3uvya5a8a1ncx.cloudfront.net"
+
+chs_url = "https://cidev.aws.chdev.org"
+
+cookie_domain = "cidev.aws.chdev.org"
+cookie_name = "__SID"
+cookie_secret = "ChGovUk-XQrbf3sLj2abFxIY2TlapsJ"
```
Comment (amartin7663) [inline terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:16]: This is a secret and should be in the vault

File: terraform/groups/ecs-service/locals.tf (hunk 1 above)
Comment (myadavch) [reply terraform/groups/ecs-service/locals.tf:3]: Fixed to `filing-maintain-service`

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars (hunk 2 above)
Comment (myadavch) [reply terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:16]: Removed from both, dev and staging.

File: terraform/groups/ecs-service/locals.tf [hunk 3]
```diff
@@ -0,0 +1,86 @@
+# Define all hardcoded local variable and local variables looked up from data resources
+locals {
+  stack_name                = "filing-maintain-service" # this must match the stack name the service deploys into
```
Comment (amartin7663) [inline terraform/groups/ecs-service/locals.tf:3]: It should be "filing-maintain". without the service

File: terraform/groups/ecs-service/data.tf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+data "aws_iam_role" "ecs_cluster_iam_role" {
+  name = "${local.name_prefix}-ecs-task-execution-role"
+}
+
+data "aws_lb" "company_requests_lb" {
+  name = "${var.environment}-chs-chgovuk"
+}
+
+data "aws_lb_listener" "company_requests_lb_listener" {
+  load_balancer_arn = data.aws_lb.company_requests_lb.arn
```
Comment (amartin7663) [inline terraform/groups/ecs-service/data.tf:33]: also change here

File: terraform/groups/ecs-service/main.tf [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  # Environmental configuration
+  environment             = var.environment
+  aws_region              = var.aws_region
+  vpc_id                  = data.aws_vpc.vpc.id
+  ecs_cluster_id          = data.aws_ecs_cluster.ecs_cluster.id
+  task_execution_role_arn = data.aws_iam_role.ecs_cluster_iam_role.arn
+  use_fargate             = false
+
+  # Load balancer configuration
+  lb_listener_arn           = data.aws_lb_listener.company_requests_lb_listener.arn
```
Comment (amartin7663) [inline terraform/groups/ecs-service/main.tf:43]: And change here

File: terraform/groups/ecs-service/data.tf [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+data "aws_ecs_cluster" "ecs_cluster" {
+  cluster_name = "${local.name_prefix}-cluster"
+}
+
+data "aws_iam_role" "ecs_cluster_iam_role" {
+  name = "${local.name_prefix}-ecs-task-execution-role"
+}
+
+data "aws_lb" "company_requests_lb" {
```
Comment (amartin7663) [inline terraform/groups/ecs-service/data.tf:28]: and here

**Review by amartin7663 — APPROVED** [review]

**Pull request merged**